### PR TITLE
Use __qualname__ as the default name

### PR DESCRIPTION
**Pull request by ozooxo**
Use https://docs.python.org/3/library/stdtypes.html#definition.__qualname__, so if we use `@circuit` inside of a class with a popular method name (such as `get()`) we no longer need to define a customized `name`.

It also better supports inner method names.

P.S The CI failure seems irrelevant -- `from contextlib import redirect_stdout, redirect_stderr
ImportError: cannot import name redirect_stdout`

### CODE REVIEW

File: circuitbreaker.py [hunk 1]
```diff
@@ -62,7 +62,10 @@ def decorate(self, function):
         Applies the circuit breaker to a function
         """
         if self._name is None:
-            self._name = function.__name__
+            try:
+                self._name = function.__qualname__
+            except AttributeError:
+                self._name = function.__name__
```
Comment (ozooxo) [inline circuitbreaker.py:68]: Unfortunately `__qualname__` in only available for python version >=3.3, so we need to fallback to the original case for a lower python version.

```
import functools
    
def print_name(f):
    @functools.wraps(f)
    def wrap(*args, **kwargs):
        try:
            print(f.__qualname__)
        except AttributeError:
            print(f.__name__)
        return f(*args, **kwargs)
    return wrap

class A:
    @print_name
    def b(self):
        pass

a = A()
a.b()

print("========")

@print_name
def c():
    pass

c()
```

```
$ python3 sandbox.py
A.b
========
c

$ python sandbox.py
b
========
c
```

**Comment by fabfuel** [comment]
Hey @ozooxo, I have replaced Travis CI with GitHub actions, which now successfully builds the pipelines again. Would you please merge the `develop` branch into your branch to check if the build passes for your PR, too?

Thanks a lot,
Fabian

**Comment by fabfuel** [comment]
Hey @ozooxo, do you want to trigger a new build? ☝️ Thanks! Fabian

**Comment by ozooxo** [comment]
@fabfuel Sorry for the delay. Merged upstream. Looks like it needs your approval to trigger the CI build. Thanks!

**Comment by fabfuel** [comment]
Thanks a lot for your contribution @ozooxo !

**Comment by ozooxo** [comment]
@fabfuel you are welcome.

**Pull request merged**